rajawali/src/main/java/c/org/rajawali3d/engine/EngineImpl.java

```diff
@@ -35,76 +33,90 @@
  * @author Jared Woolston (Jared.Woolston@gmail.com)
  */
 @NotThreadSafe
-public class RendererImpl implements Renderer, ISurfaceRenderer {
+public class EngineImpl implements Engine, ISurfaceRenderer {
```

Comment:
Yea. For the time being I dont think it makes sense to worry about an abstract implementation. There is still a fair bit of research on Vulkan to do before we get there.